plugins/twostepauth.php

```diff
@@ -971,7 +972,7 @@ function twostepauth_admin_error($title, $msg)
 function twostepauth_authorize_ip($ip, $uid, $code) {
     global $db;
     $loc = twostepauth_get_location($ip);
-    $db->insert_query("twostepauth_authorizations", array("ip" => $db->escape_string($ip), "location" => $db->escape_string($loc), "code" => $db->escape_string($code), "uid" => $db->escape_string($uid)));
+    $db->insert_query("twostepauth_authorizations", array("ip" => $db->escape_string($ip), "location" => $db->escape_string($loc), "code" => $db->escape_string($code), "username" => $db->escape_string($uid)));
```

Comment:
Yes but in this particular line you are assuming `$uid` is always going to be an `username`, which is not true if you see the line I linked.